InaraResponder/InaraResponder.cs

```diff
@@ -658,41 +664,47 @@ private void handleShipSwappedEvent(ShipSwappedEvent @event)
         {
             if (!string.IsNullOrEmpty(@event.storedship))
             {
-                Ship storedShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.storedshipid);
-                Dictionary<string, object> storedShipData = new Dictionary<string, object>()
+                var storedShipData = new Dictionary<string, object>()
                 {
-                    { "shipType", storedShip.EDName },
-                    { "shipGameID", storedShip.LocalId },
-                    { "shipName", storedShip.name },
-                    { "shipIdent", storedShip.ident },
-                    { "isHot", storedShip.hot },
-                    { "shipRole", storedShip.Role.invariantName },
-                    { "isCurrentShip", true },
+                    { "shipType", @event.storedEdModel },
+                    { "shipGameID", @event.storedshipid },
+                    { "isCurrentShip", false },
                     { "starsystemName", EDDI.Instance.CurrentStarSystem?.systemname },
                     { "stationName", EDDI.Instance.CurrentStation?.name },
                     { "marketID", EDDI.Instance.CurrentStation?.marketId }
                 };
+                var storedShip = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship Monitor")).GetShip(@event.storedshipid);
+                if (storedShip?.EDName == @event.storedEdModel)
+                {
+                    storedShipData.Add("shipName", storedShip?.name);
```

Comment:
We can remove the `?`s from `storedShip`.

Reply:
No. The `GetShip` method can return null a null value if the local ship id is null or not found int he shipyard.